diff --git a/src/Nodes/DocumentNode.php b/src/Nodes/DocumentNode.php
--- a/src/Nodes/DocumentNode.php
+++ b/src/Nodes/DocumentNode.php
@@ -279,6 +279,11 @@ public function getFootnoteTargetAnonymous(): FootnoteTarget|null
         return null;
     }

+    public function hasDocumentEntry(): bool
+    {
+        return $this->documentEntry !== null;
+    }
+
     public function getDocumentEntry(): DocumentEntryNode
     {
         if ($this->documentEntry === null) {
